src/backend/joanie/signature/backends/base.py

```diff
@@ -0,0 +1,69 @@
+"""Base Signature Backend"""
+
+
+class BaseSignatureBackend:
+    """
+    The Signature base class.
+    It contains generic methods to trigger
+    the workflow process of signing a document,
+    and create the invitation link.
+    """
+
+    name = "base"
+
+    def __init__(self, configuration=None):
+        self.configuration = (
+            configuration.get("configuration", {}) if configuration else {}
+        )
+
+    def upload_file(self, *args, **kwargs):
+        """
+        Wrapper to start a signature procedure with a specifc amount
+        of required signers.
+        This method implies : to prepare a workflow, to upload
+        a document, and start the process to sign them.
+        """
+        raise NotImplementedError(
+            "subclasses of BaseSignatureBackend must provide a upload_file() method."
+        )
+
+    def create_invitation_link(self, recipient_email: str, *args, **kwargs):
+        """
+        Create invitation link to sign the document for a given email recipient.
+        It will bring the user to sign every document linked to his email address.
+        """
+        raise NotImplementedError(
+            "subclasses of BaseSignatureBackend must provide a create_invitation_link() method."
+        )
+
+    def sign_selection_of_document(self, recipient_email: str, workflow_ids: list):
+        """
+        Prepare a specific invitation link that wraps more than one document to sign at once.
```

Comment:
if it prepares an invitation link a better name would be something like `get_signature_invitation_link`?

Reply:
It prepares a "one-time" invitation link for a selection of documents to sign. I have just applied the recommended changes.